quickwit/quickwit-storage/src/storage_resolver.rs

```diff
@@ -51,8 +51,7 @@ impl StorageResolver {
         StorageResolverBuilder::default()
     }
 
-    /// Resolves the given URI.
-    pub async fn resolve(&self, uri: &Uri) -> Result<Arc<dyn Storage>, StorageResolverError> {
+    fn get_storage_factory(&self, uri: &Uri) -> Result<&dyn StorageFactory, StorageResolverError> {
```

Comment:
What is the benefit?

Reply:
this is work in progress, I'm looking for ways to properly disable the security in the LocalFileStorage only for the FileSource. One way will be to have a specific resolve function that creates a LocalFileStorage with a flag that disables the check.

Reply:
I'll actually revert this, I found a simpler way around the check by splitting the URI into the directory + the filename in the source. The tests now go through without disabling the security check!